Makefile.ffmpeg

```diff
@@ -230,7 +204,9 @@ ifeq (yes,$(CONFIG_LIBX264))
 
 EXTLIBS  += libx264
 ENCODERS += libx264
+ifneq (yes,$(CONFIG_FFMPEG_STATIC))
```

Comment:
Don't do this, if ffmpeg isn't compiled statically this isn't even used....

The patch needs to be understood and adapted, especially why it exists. Adapting it is easy, but why is it there in the first place?

Reply:
was only for testing ... I will delete the DIFF if it pass.

Reply:
It does pass if we remove that line ... like you pointed out: why was added?
Let me investigate more :( .

Reply:
I am far from specialist in patches ... but is this not intended to be used only with ffmpeg 3.1.2?

![patch](https://user-images.githubusercontent.com/4031821/229655074-7ad98cda-d8df-4b34-884b-742b46b89ed5.png)

Reply:
I was wrong ... I can see is applied also for ffmpeg4:
![patch_4](https://user-images.githubusercontent.com/4031821/229664191-2b310d34-9020-447f-add6-037b26fd3e27.png)

I am afraid that is way over my capability. I never done any patches ... I will definitely need help.

Reply:
It's a single change, really easy and perfect to learn. It's likely the offset that's just too high now. It was designed for ffmpeg 3 though apparently.

I still wonder why it exists in the first place.

Reply:
encouraged by you, I decided to try. The problem was that line that has to be removed, changed (you will see in changes). The offset somehow is compensated by patcher (for ffmpeg4). I cleaned up the patch for ffmpeg 5.1.2. I can confirm now is adding the diff (and is not generating the libx264.c.rej file during compilation). Let's me know what do you think.

> I still wonder why it exists in the first place.

This is way over my league :( . I can tell you that diff file was added sometime in 2017 (at the starting of transcoding) with commit: https://github.com/tvheadend/tvheadend/commit/7840d91ce3d5e0f4d919431092cef71be08f1873
Why was added? I have no idea ... I can tell you that now, if i search after 'x264profile' in a project compiled (with all libraries downloaded) is not found anywhere else.
![ffmpeg5](https://user-images.githubusercontent.com/4031821/229986076-910701f9-45f3-4b31-b29f-12c0a55e6c2d.png)